components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/bean/ScopeTest.java

```diff
@@ -0,0 +1,119 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.identity.oauth2.bean;
+
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertNotEquals;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+@PrepareForTest(StringBuilder.class)
+public class ScopeTest {
+    Scope scope;
+    String name = "readClaims";
+    String description = "Test cases to test scope";
+
+    ArrayList<String> bindings = new ArrayList<>(Arrays.asList("scope1", "scope2"));
+
+    @BeforeTest
+    public void setUp() throws IllegalAccessException, InstantiationException {
+        scope = new Scope(name, description);
+        scope = new Scope(name, description, bindings);
+    }
+
+
+    @Test
+    public void testGetName() throws Exception {
+        assertEquals(scope.getName(),
+                "readClaims", "Valid getName().");
+        assertNotEquals(scope.getName(),
+                "ReadClaims", "Invalid getName().");
+    }
+
+    @Test
+    public void testSetName() throws Exception {
+        scope.setName(name = "value");
+        assertEquals(scope.getName(),
+                "value", "Valid getName().");
+        assertNotEquals(scope.getName(),
+                "readClaims", "Invalid getName().");
+
+    }
+
+    @Test
+    public void testGetDescription() throws Exception {
+        assertEquals(scope.getDescription(),
+                "Test cases to test scope", "Valid getDescription().");
+        assertNotEquals(scope.getDescription(),
+                "Test cases to testscope", "Invalid getDescription().");
+    }
+
+    @Test
+    public void testSetDescription() throws Exception {
+        scope.setDescription(name = "Testing authcontext oauth scope");
+        assertEquals(scope.getDescription(),
+                "Testing authcontext oauth scope", "Valid getDescription().");
+        assertNotEquals(scope.getDescription(),
+                "Testing authcontextoauth scope", "Invalid getDescription().");
+    }
+
+
+    @Test
+    public void testAddBindings() throws Exception {
+        scope.addBinding("scope3");
+        assertTrue(scope.getBindings().contains("scope3"));
+
+    }
+
+    @Test
+    public void testAddBinding() throws Exception {
+
+        scope.addBindings(Arrays.asList(new String[]{"scope3", "scope4"}));
+        assertTrue(scope.getBindings().containsAll(Arrays.asList(new String[]{"scope3", "scope4"})));
+    }
+
+    @Test
+    public void testGetBindings() throws Exception {
+        assertNotNull(bindings);
+        bindings.add("scope4");
+        assertEquals(scope.getBindings(), bindings, "Actual size is not match with expected one");
```

Comment:
Actual size is not match with expected one --> Actual size does not match with expected one

Reply:
Fixed